src/hooks/useEventEmitter.ts

```diff
@@ -54,12 +66,22 @@ export function useEventEmitter(
             // Create event listener that calls handler function stored in ref
             const eventListener = (...args) => savedHandler.current(...args);
 
+            // XXX: We cannot use instanceof as that would break tests
             // Add event listener
-            emitter.on(eventName, eventListener);
+            if ((emitter as EventTarget).addEventListener && typeof eventName === "string") {
+                (emitter as EventTarget).addEventListener(eventName as string, eventListener);
```

Comment:
Can we use type guards here to clean it up? 
https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Reply:
Cool - didn't know TS had this!